packages/edit-post/src/editor.js

```diff
@@ -129,22 +129,27 @@ function Editor( {
 			keepCaretInsideBlock,
 		};
 
+		// Defer to passed setting for `allowedBlockTypes` if provided as
+		// anything other than `true` (where `true` is equivalent to allow
+		// all block types).
+		const defaultAllowedBlockTypes =
+			true === settings.allowedBlockTypes
+				? map( blockTypes, 'name' )
+				: settings.allowedBlockTypes || [];
 		// Omit hidden block types if exists and non-empty.
 		if ( size( hiddenBlockTypes ) > 0 ) {
-			// Defer to passed setting for `allowedBlockTypes` if provided as
-			// anything other than `true` (where `true` is equivalent to allow
-			// all block types).
-			const defaultAllowedBlockTypes =
-				true === settings.allowedBlockTypes
-					? map( blockTypes, 'name' )
-					: settings.allowedBlockTypes || [];
-
 			result.allowedBlockTypes = without(
 				defaultAllowedBlockTypes,
 				...hiddenBlockTypes
 			);
 		}
-
+		// Hide `Template Part` when we are not in template mode.
+		if ( ! isTemplateMode ) {
+			result.allowedBlockTypes = without(
```

Comment:
After starting the changes to introduce an editor setting `__unstableDisallowedBlockTypes`, it seems that requires many changes and IMO adds much more complexity for now, since we need a well thought API for handling disallowing blocks per context(and whatever this context will be).

@youknowriad do you have strong opinion about this? I can finish them and push to check if you want.

Reply:
The simpler, the better for me :)